test/integration/EndToEnd.t.sol

```diff
@@ -45,44 +134,151 @@ contract TestEndToEnd is Test {
+            poolId,
+            scId,
+            cv.centrifugeId,
+            address(cv.poolManager).toBytes32(),
+            MessageLib.UpdateContractVaultUpdate({
+                vaultOrFactory: vaultFactory.toBytes32(),
+                assetId: assetId.raw(),
+                kind: uint8(VaultUpdateKind.DeployAndLink)
+            }).serialize()
+        );
     }
 
     /// forge-config: default.isolate = true
-    function testConfigurePool(bool sameChain) public {
-        (HubDeployer ch, VaultsDeployer cv) = _getDeploys(sameChain);
-        uint16 cvChainId = cv.messageDispatcher().localCentrifugeId();
+    function testAsyncDeposit(bool sameChain) public {
+        _setCV(sameChain);
+        (PoolId poolId, ShareClassId scId, AssetId assetId) = _configurePool(address(cv.asyncVaultFactory));
+
+        IShareToken shareToken = IShareToken(cv.poolManager.shareToken(poolId, scId));
+        (address asset,) = cv.poolManager.idToAsset(assetId);
+        IAsyncVault vault = IAsyncVault(shareToken.vault(address(asset)));
 
-        IGuardian guardian = ch.guardian();
-        AssetId usd = deployA.USD();
-        vm.prank(address(guardian.safe()));
-        PoolId poolId = guardian.createPool(1, FM, usd);
+        vm.startPrank(INVESTOR_A);
+        ERC20(asset).approve(address(vault), INVESTOR_A_AMOUNT);
+        vault.requestDeposit(INVESTOR_A_AMOUNT, INVESTOR_A, INVESTOR_A);
 
         vm.startPrank(FM);
-        ch.hub().notifyPool{value: GAS}(poolId, cvChainId);
+        IERC7726 valuation = h.holdings.valuation(poolId, scId, assetId);
+        h.hub.approveDeposits{value: GAS}(poolId, scId, assetId, INVESTOR_A_AMOUNT, valuation);
+        h.hub.issueShares(poolId, scId, assetId, IDENTITY_PRICE);
+
+        vm.startPrank(ANY);
+        h.hub.claimDeposit{value: GAS}(poolId, scId, assetId, INVESTOR_A.toBytes32());
+
+        //vault.mint(INVESTOR_A_AMOUNT, INVESTOR_A);
+
+        //assertEq(shareToken.balanceOf(INVESTOR_A), INVESTOR_A_AMOUNT);
+    }
+
+    /// forge-config: default.isolate = true
+    function testSyncDeposit(bool sameChain) public {
+        _setCV(sameChain);
+        (PoolId poolId, ShareClassId scId, AssetId assetId) = _configurePool(address(cv.syncDepositVaultFactory));
+
+        IShareToken shareToken = IShareToken(cv.poolManager.shareToken(poolId, scId));
+        (address asset,) = cv.poolManager.idToAsset(assetId);
+        SyncDepositVault vault = SyncDepositVault(shareToken.vault(address(asset)));
+
+        vm.startPrank(INVESTOR_A);
+        ERC20(asset).approve(address(vault), INVESTOR_A_AMOUNT);
+        vault.deposit(INVESTOR_A_AMOUNT, INVESTOR_A);
+
+        // TODO: Continue investing process
+        //cv.balanceSheet.approveDeposits(poolId, scId, assetId, INVESTOR_A_AMOUNT);
+        //cv.balanceSheet.issue(poolId, scId, assetId, INVESTOR_A_AMOUNT);
 
-        assert(cv.poolManager().pools(poolId) != 0);
+        //vault.mint(INVESTOR_A_AMOUNT, INVESTOR_A);
```

Comment:
Nit: I think these commented out steps are wrong but we can remove them after merging this PR.

* The sync vault deposit already issues shares for the investor
* `cv.balanceSheet.approveDeposits` should never be called manually as it is a reaction to a deposit approval on the hub site

Reply:
Thanks for the guiding!